cli: Replace `sucrase` with `swc` for transpilation (#10394)
backstage/backstage: .changeset/rotten-books-crash.md

---
'@backstage/dev-utils': patch
---

> Removing dependency on `react-hot-loader`, the we ship `swc` and `React Refresh` out of the box now with the `cli`

Reviewer: ```suggestion
Removed the dependency and setup of `react-hot-loader`, since the `@backstage/cli` now uses `swc` with `React Refresh` instead.
```
